flyteplugins/go/tasks/pluginmachinery/core/template/template.go

```diff
@@ -199,6 +201,19 @@ func serializeLiteralScalar(l *idlCore.Scalar) (string, error) {
 		return o.Blob.Uri, nil
 	case *idlCore.Scalar_Schema:
 		return o.Schema.Uri, nil
+	case *idlCore.Scalar_Binary:
+		binaryBytes := o.Binary.Value
+		var currVal any
+		if o.Binary.Tag == coreutils.MESSAGEPACK {
+			err := msgpack.Unmarshal(binaryBytes, &currVal)
+			if err != nil {
+				return "", fmt.Errorf("failed to unmarshal messagepack bytes with literal:[%v], err:[%v]", l, err)
+			}
+			// TODO: Try to support Primitive_Datetime, Primitive_Duration, Flyte File, and Flyte Directory.
+			return fmt.Sprintf("%v", currVal), nil
+		}
+		return "", fmt.Errorf("unsupported binary tag [%v]", o.Binary.Tag)
```

Comment:
In this PR, datetime, timedelta, flyte directory, and flyte file are not supported, and these are not supported before, so let's ignore them now.

The hard part of this is that we have only `scalar` value, but not expected literal type or expected python type as out hint, that's why I can't support in this PR.

cc @pingsutw @wild-endeavor @eapolinario